### PR TITLE
feat(Set Node): Overhaul

### PR DESCRIPTION
**Pull request by michael-radency**
Github issue / Community forum post (link here to close automatically): https://github.com/n8n-io/n8n/pull/6348

### CODE REVIEW

**Comment by luizeof** [comment]
it would be nice to have an option to remove some fields

**Comment by luizeof** [comment]
This pull is fantastic, is there any preview date or version?

As it changes the node interface, an ETA would be nice so we have time to show it to the community to avoid adoption issues for less experienced users.

File: packages/workflow/src/NodeHelpers.ts [hunk 1]
```diff
@@ -1169,12 +1170,16 @@ export const tryToParseBoolean = (value: unknown): value is boolean => {
 		return value.toLowerCase() === 'true';
 	}
 
-	const num = Number(value);
-	if (num === 0) {
-		return false;
-	} else if (num === 1) {
-		return true;
+	// If value is not a empty string, try to parse it to a number
+	if (!(typeof value === 'string' && value.trim() === '')) {
```
Comment (MiloradFilipovic) [inline packages/workflow/src/NodeHelpers.ts:1174]: Can this be turned into `if(String(val) !== '')`?

**Review by MiloradFilipovic — CHANGES_REQUESTED** [review]
I left a minor comment in the code but am also getting some errors/inconsistencies when using the node:

1. When using `number` type and entering something that is not a number, I am getting `null` value instead of validation error:
image
3. I am unable to delete fields, getting this console error when clicking thrash icon:
image

**Review by maspio — DISMISSED** [review]

**Review by MiloradFilipovic — DISMISSED** [review]
Looking good, thanks for addressing previous comments, @michael-radency 🙌
There is one minor thing that I noticed but ultimately it's a product decision:
When Ignore type conversion is turned on, inline validation errors (warnings next to inputs) are still showing up which could look confusing to users:
image

**Review by maspio — DISMISSED** [review]

**Review by maspio — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.8.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.8.1)

**Comment by artemdzyuba** [comment]
In 1.8.2, this Set node messes with null and undefined values. See [this thread](https://community.n8n.io/t/null-and-undefined-in-string-fields/31060).

**Pull request merged**